src/Shared/RoslynUtils/WellKnownTypes.cs

```diff
@@ -86,6 +86,26 @@ private INamedTypeSymbol GetAndCache(int index)
         return _lazyWellKnownTypes[index]!;
     }
 
+    // Filter for types within well-known (framework-owned) assemblies only.
+    private INamedTypeSymbol? GetTypeByMetadataNameInTargetAssembly(string metadataName)
+    {
+        var types = _compilation.GetTypesByMetadataName(metadataName);
+        if (types.Length == 1)
```

Comment:
Should we quickly return null if length is 0?

Reply:
Doesn't hurt to!